internal/planner/mapper/mapper.go

```diff
@@ -413,25 +413,23 @@ func resolveAggregates(
 				childMapping = childMapping.CloneWithoutRender()
 				mapping.SetChildAt(index, childMapping)
 
-				if !childIsMapped {
```

Comment:
this was a nasty bug that took many hours to spot and fix.
It because visible for `TestOneToManyToOneWithSumOfDeepFilterSubTypeOfBothDescAndAsc` test where 2 aggregates are used on the same related object:
```graphql
query {
    Author {
        name
        s1: _sum(book: {field: rating, filter: {publisher: {yearOpened: {_eq: 2013}}}})
        s2: _sum(book: {field: rating, filter: {publisher: {yearOpened: {_ge: 2020}}}})
    }
}
```
After the first aggregate is processed with correct `[]Requestable` fields, the second would reuse the results of the frst which doesn't include resolving of fields. So `Fields` field would have a `nil` value. Which happened to be interpreted by the fetcher as all-fields-requested.
So when dealing with secondary indexes I explicitly added to the `fields` array a relation id field (like `book_id`). That made the fetcher to fetch only the related id field and would prevent the filter from letting `yearOpened: {_ge: 2020}` condition to pass, because it's not fetched.

That's why filter dependencies should be resolved for every aggregate.

Let me know if you think there is a better solution.